### PR TITLE
Deprecate KindOverride fields and use internal api group.

### PR DESCRIPTION
**Pull request by jiachengxu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Deprecate KindOverride fields and use internal api group.
```

### CODE REVIEW

File: pkg/manager/internal/controllers/customresourcediscovery_controller_test.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func TestCustomResourceDiscoveryReconciler(t *testing.T) {
 
 	internalCRD := &apiextensionsv1.CustomResourceDefinition{}
 	require.NoError(t, r.Client.Get(ctx, types.NamespacedName{
-		Name: strings.Join([]string{"redis", serviceClusterName, crDiscovery.Namespace}, "."),
+		Name: strings.Join([]string{"redis", "internal", serviceClusterName, crDiscovery.Namespace}, "."),
```
Comment (aborilov) [inline pkg/manager/internal/controllers/customresourcediscovery_controller_test.go:138]: why not using constant here?

File: test/integration/apiserver.go [hunk 2]
```diff
@@ -380,14 +378,14 @@ func instanceService(ctx context.Context, conn *grpc.ClientConn, tenantAccount *
 		offering := &catalogv1alpha1.Offering{
 			ObjectMeta: metav1.ObjectMeta{
 				Namespace: tenantAccount.Status.Namespace.Name,
-				Name:      strings.Join([]string{"dbs", serviceCluster.Name, providerAccount.Name}, "."),
+				Name:      strings.Join([]string{"dbs", "external", serviceCluster.Name, providerAccount.Name}, "."),
```
Comment (aborilov) [inline test/integration/apiserver.go:381]: same here

File: pkg/manager/internal/controllers/customresourcediscovery_controller_test.go (hunk 1 above)
Comment (jiachengxu) [reply pkg/manager/internal/controllers/customresourcediscovery_controller_test.go:138]: fixed here.

File: test/integration/apiserver.go (hunk 2 above)
Comment (jiachengxu) [reply test/integration/apiserver.go:381]: We cannot use constants pkg in test, since constants are under `internal` package.

Comment (aborilov) [reply test/integration/apiserver.go:381]: that is sad

Comment (jiachengxu) [reply test/integration/apiserver.go:381]: hmm, if needed, we can move the whole constants package out of internal, what do you think?

Comment (aborilov) [reply test/integration/apiserver.go:381]: it's not a big deal, I'm ok to merge it as is, but yeah, I think constants should not be in the internals

Comment (jiachengxu) [reply test/integration/apiserver.go:381]: OK, let's do it in a dedicated PR and move the whole constants pkg out of internal

Comment (aborilov) [reply test/integration/apiserver.go:381]: 👍

**Pull request merged**